lib/ansible/modules/system/hostname.py

```diff
@@ -374,7 +376,7 @@ class SystemdStrategy(GenericStrategy):
     """
 
     def get_current_hostname(self):
-        cmd = ['hostname']
+        cmd = [self.hostname_cmd, '--transient', 'status']
```

Comment:
the options are not compatible, set `self.hostnamectl_cmd`, use that and appropriate options when populated, default to previous code when not